src/spyglass/spikesorting/analysis/v1/group.py

```diff
@@ -170,6 +176,10 @@ def fetch_spike_data(
             sorting_spike_times = nwb_file[nwb_field_name][
                 "spike_times"
             ].to_list()
+            file_unit_ids = [
+                f"{merge_id}_{unit_number}"
```

Comment:
I think these should be preserved as a tuple and later stored as separate fields. I think I'm seeing some places where the ordinal value of unit number is used for some error checks. Concatenation erodes your ability to do this. `uuids` should also be treated as 'random', meaning that we can't expect a future ability so split these things. I don't think our uuid protocol permits underscores, but I'd hate to be proven wrong.

Reply:
It seems like this could even be returned as a dict to cut memory costs `{merge_id: List[unit]}`

Reply:
I think the memory is pretty marginal here.  What I really wanted from this was a per-unit identifier that could be zipped together with the list of spike times to make tracking units easier.  I think the dictionary proposal I put above is a middle ground between concatenating and this solution